test/test_nn.py

```diff
@@ -13301,47 +13301,50 @@ def test_conv3d_valid_padding_backward(self, device):
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='slow3d_dilated'),
         subtest(((2, 6, 7, 8, 9), True, True, 3, torch.strided, torch._C._ConvBackend.SlowTranspose3d),
-                decorators=[onlyCPU, disableMkldnn], name='slow3d_dilated_transposed'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='slow3d_dilated_transposed'),
         subtest(((0, 6, 7), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_batch1d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_batch1d'),
         subtest(((2, 0, 7), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_channel1d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_channel1d'),
         subtest(((0, 0, 7), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_batch_channel1d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_batch_channel1d'),
         subtest(((0, 6, 7, 8), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_batch2d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_batch2d'),
         subtest(((2, 0, 7, 8), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_channel2d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_channel2d'),
         subtest(((0, 0, 7, 8), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_batch_channel2d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_batch_channel2d'),
         subtest(((0, 6, 7, 8, 9), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_batch3d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_batch3d'),
         subtest(((2, 0, 7, 8, 9), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_channel3d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_channel3d'),
         subtest(((0, 0, 7, 8, 9), False, False, 3, torch.strided, torch._C._ConvBackend.Empty),
-                decorators=[onlyCPU, disableMkldnn], name='empty_batch_channel3d'),
+                decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN], name='empty_batch_channel3d'),
```

Comment:
I'm confused, why didn't we disable cudnn before?

Reply:
Good question - I wrongly restricted the slow tests to CPU-only before. Now they run on CUDA as well (`onlyCPU` -> `onlyNativeDeviceTypes`), which means `disablecuDNN` should be set to ensure the slow algos are selected.